.goreleaser.yaml

```diff
@@ -0,0 +1,522 @@
+# This is an example .goreleaser.yml file with some sensible defaults.
+# Make sure to check the documentation at https://goreleaser.com
+
+# The lines below are called `modelines`. See `:help modeline`
+# Feel free to remove those if you don't want/need to use them.
+# yaml-language-server: $schema=https://goreleaser.com/static/schema.json
+# vim: set ts=2 sw=2 tw=0 fo=cnqoj
+
+project_name: bao
+
+before:
+  hooks:
+    # You may remove this if you don't use go modules.
+    - go mod tidy
+    # you may remove this if you don't need go generate
+    - go generate ./...
+    - make bootstrap
+
+builds:
+  - id: builds-linux
+    binary: "dist/{{ .Os }}/{{ .Arch }}/{{ .ProjectName }}"
+    ldflags:
+      - -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{ .CommitDate }}
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+    goarch:
+      - amd64
+      - arm
+      - arm64
+      - riscv64
+    goarm:
+      - "6"
+    no_unique_dist_dir: true
+    mod_timestamp: "{{ .CommitTimestamp }}"
+    skip: false
+  - id: builds-other
+    ldflags:
+      - -X main.version={{.Version}} -X main.commit={{.Commit}} -X main.date={{ .CommitDate }}
+    env:
+      - CGO_ENABLED=0
+    goos:
+      - darwin
+      - dragonfly
+      - freebsd
+      - illumos
+      - netbsd
+      - openbsd
+      - solaris
+      - wasip1
+      - windows
+    goarch:
+      - amd64
+      - arm
+      - arm64
+      - riscv64
+      - wasm
+    goarm:
+      - "6"
+      - "7"
+    ignore:
+      - goos: darwin
+        goarm: "6"
+      - goos: darwin
+        goarch: arm
+      - goos: darwin
+        goarch: riscv64
+      - goos: darwin
+        goarch: wasm
+      - goos: dragofly
+        goarch: arm
+      - goos: dragofly
+        goarch: arm64
+      - goos: dragofly
+        goarch: riscv64
+      - goos: dragofly
+        goarch: wasm
+      - goos: freebsd
+        goarm: "7"
+      - goos: freebsd
+        goarch: wasm
+      - goos: illumos
+        goarch: arm
+      - goos: illumos
+        goarch: arm64
+      - goos: illumos
+        goarch: wasm
+      - goos: netbsd
+        goarch: riscv64
+      - goos: netbsd
+        goarch: wasm
+      - goos: netbsd
+        goarm: "7"
+      - goos: openbsd
+        goarch: riscv64
+      - goos: openbsd
+        goarch: wasm
+      - goos: openbsd
+        goarm: "7"
+      - goos: solaris
+        goarch: arm
+      - goos: solaris
+        goarch: arm64
+      - goos: solaris
+        goarch: wasm
+      - goos: windows
+        goarch: arm
+        goarm: "7"
+      - goos: windows
+        goarch: riscv64
+      - goos: windows
+        goarch: wasm
+    mod_timestamp: "{{ .CommitTimestamp }}"
+    skip: false
+
+report_sizes: true
+
+nfpms:
```

Comment:
Something causes the binary paths of the created packages to be weird. For example the `amd64` Debian package has the `bao` binary placed in `/usr/bin/dist/linux/amd64/bao` which is likely to cause issues. I think the binary should just be placed in `/usr/bin/bao`

Reply:
Ah so that is the side effect. I will need to replace our dockerfile then which will break or other ci.

Reply:
@naphelps yeah, my intent was to remove the build CI if we're happy with this.

Reply:
Reverted the directory changes in the build process. Adjusted the dockerfile to match. This will break the existing CI.